### ISSUE TITLE
Search for some words does not work

### ISSUE THREAD

**Issue by jekeam** [nbuytaert1/apex-select2#111]
I can not understand what was happening, I tried different settings, but searching for certain words does not work, what could be the problem?
Example (Containse & ignore Case):
![Снимок](https://user-images.githubusercontent.com/19660852/58400450-75a7be80-8074-11e9-8c8b-1f238cc0f517.PNG)

Not found, but the string exists if you enter another word:
![Снимок2](https://user-images.githubusercontent.com/19660852/58400457-77718200-8074-11e9-8b43-042d62588142.PNG)

**Comment by nbuytaert1** [comment]
That's odd. Is it possible to reproduce this behavior on apex.oracle.com and share developer credentials for your workspace?

**Comment by jekeam** [comment]
I don't quite understand how these options work inside, you can explain their behavior in more detail:
Contains & Ignore Case
Contains & Ignore Case, with Diacritics
Contains & Case Sensitive
Exact & Ignore Case
Exact & Case Sensitive
Starts With & Ignore Case
Starts With & Case Sensitive
Multi-word Search

Demo:
https://apex.oracle.com/pls/apex/f?p=128665:2::::::
cred.: suineg/suineg@inbox.ru/12345678

new example:
![456](https://user-images.githubusercontent.com/19660852/58453580-1cec2a80-8135-11e9-865c-5fad0001ab42.PNG)
![456546546](https://user-images.githubusercontent.com/19660852/58453588-1fe71b00-8135-11e9-97f4-742ef9bb933c.PNG)

I suspect that the problem is caused by a large number of lines in the directory, since I could not reproduce the error in a smaller directory.

![45546](https://user-images.githubusercontent.com/19660852/58453681-7b190d80-8135-11e9-81dc-64acd97a8539.PNG)

**Comment by jekeam** [comment]
The problem will disappear if you add a condition: `where name like 'VBNKRUM1VGD%'`
Does this really mean the number of lines?

**Comment by nbuytaert1** [comment]
The _Search Logic_ setting determines how the entered search term is matched with the display column in your LOV.

The problem here is indeed the large amount of results being returned to the client. When I run the page in debug mode, I notice the following error:

`PL/SQL: numeric or value error: character string buffer too small
ORA-06512: at "APEX_190100.WWV_FLOW_DYNAMIC_EXEC", line 1500
ORA-06512: at "APEX_190100.WWV_FLOW_DYNAMIC_EXEC", line 2501
ORA-06512: at line 761
ORA-06512: at line 793
ORA-06512: at "SYS.DBMS_SYS_SQL", line 2120
ORA-06512: at "SYS.WWV_DBMS_SQL_APEX_190100", line 599
ORA-06512: at "APEX_190100.WWV_FLOW_DYNAMIC_EXEC", line 2486
ORA-06512: at "APEX_190100.WWV_FLOW_DYNAMIC_EXEC", line 1476`

This happens when the plugin code tries to build the JSON response body. This response includes the matched rows after querying the LOV.

I was unable to find a solution for this problem, but there's a workaround. Simply fill in a number for the _Lazy-append Row Count_ setting on your Select2 page item (for example: 10). That will limit the amount of rows being returned to the client, and thus avoiding the above error.

**Comment by jekeam** [comment]
Yes, it helped, thanks. I did not get the idea to enable debugging, even ashamed)
As for the parameters, I understood. But I did not understand how each of them works in more detail, you can describe in more detail or send ref on doc.?

**Comment by nbuytaert1** [comment]
Contains & Ignore Case = case insensitive regular search
Contains & Ignore Case, with Diacritics = case insensitive regular search, not taking into account special characters (for example: searching for "aero" will return "Aeróbics")
Contains & Case Sensitive = case sensitive regular search
Exact & Ignore Case = the search term must match the search option completely (case insensitive)
Exact & Case Sensitive = the search term must match the search option completely (case sensitive)
Starts With & Ignore Case = like 'search term%' (case insensitive)
Starts With & Case Sensitive = like 'search term%' (case sensitive)
Multi-word Search = see https://stackoverflow.com/questions/54763047/search-for-multiple-words-in-an-option-in-select2
